Frontend/src/Components/Graph/CustomGraph.js

```diff
@@ -1,427 +1,496 @@
+  }
+}
+
+// the options fed into the graph object, save regardless of datasets
+function getOptions(now, secondsRetained) {
+  // console.log("now:", now);
+  // console.log("diff:", DateTime.now().diff(now).toHuman());
+
+  return {
+    responsive: true,
+    maintainAspectRatio: false,
+    indexAxis: "x",
+    animation: false,
+    plugins: {
+      // unknown if decimation is functional
+      // decimation: {
+      //   enabled: true,
+      //   algorithm: "min-max",
+      // },
+      legend: {
+        display: false,
+      },
+      tooltip: {
+        callbacks: {
+          label: (item) => {
+            // console.log(item);
+
+            // custom dataset label
+            // name: value unit
+            return `${item.dataset.label}: ${item.formattedValue}${
+              units[item.dataset.key]
+            }`;
+          },
+        },
+      },
+    },
+    scales: {
+      x: {
+        type: "time",
+        reverse: true,
+        // ticks: {
+        //   source: "data",
+        // },
+        bounds: "data", // ticks?
+        time: {
+          unit: "second",
+          stepSize: 1,
+          tooltipFormat: "h:mm:ss a",
+          displayFormats: {
+            second: "h:mm:ss a",
+          },
+        },
+
+        // show the last secondsRetained seconds
+        max: now.toISO(),
+        min: now.minus(secondsRetained * 1000).toISO(),
+      },
+      y: {
+        suggestedMin: 0,
+      },
+    },
+    elements: {
+      line: {
+        borderCapStyle: "round",
+        borderJoinStyle: "round",
+      },
+    },
+    interaction: {
+      mode: "nearest",
+      axis: "x",
+      intersect: false,
+    },
+    datasets: {
+      line: {
+        pointBackgroundColor: "transparent",
+        pointBorderColor: "transparent",
+        pointRadius: 10,
+      },
+    },
+  };
+}
+
+/**
+ * Creates a customizable graph
+ *
+ * @param {any} props the props to pass to this graph; any {@link StackProps} will be passed to the overarching container
+ * @param {(name: string, isNew: boolean, datasets: string[]) => void} props.onSave the callback function for when the user attempts to save this graph
+ * @param {string} props.title the title that this graph has
+ * @param {{category: string, values: Dataset[]}[]} props.categories the color and other associated data for each dataset, organized in an object
+ * @param {Dataset[]} props.allDatasets a list of all datasets, complete with ID and color information
+ * @param {string[]} props.initialDatasets a list of the IDs of the initial datasets
+ * @param {any} props.packedData the queue of data, packed in Chart.js format, coming from the solar car
+ * @param {number} props.secondsRetained the number of seconds to retain a point on this grpah
+ * @returns
+ */
+export default function CustomGraph(props) {
+  // destructure props
+  const {
+    onSave,
+    title,
+    categories,
+    packedData,
+    initialDatasets,
+    allDatasets,
+    secondsRetained,
```

Comment:
categories and allDatasets aren't used. secondsRetained also isn't passed

Reply:
`secondsRetained` isn't passed since I forgot to include it in the data of the graphs to be saved. I'll add that in last after the other changes.

Reply:
Added `secondsRetained` into the stored graph state. All done